### PR TITLE
ACMS-1389: Updated content type templates for Article, Person, Place, Event, Page for Media.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1389

**Proposed changes**
- Rewrite Image field to include DAM media.
- Updated content type templates for Article, Person, Place, Event, Page for Media(CORE, DAM).

**Alternatives considered**
NA

**Testing steps**
- Use this branch
- Install site using `acms site:install`
- Install ACMS DAM module `drush in acquia_cms_dam`
- Login into application
- Create node using any content type
- Select DAM media for image in first node
- Select CORE media for image in second node
- Verify images rendering fine on both nodes

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_dam/acquia_cms_dam.module [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if ($fieldConfig) {
+      _acquia_cms_dam_update_image_field($fieldConfig);
+    }
+  }
+}
+
+/**
+ * Helper function to update content type image field.
+ */
+function _acquia_cms_dam_update_image_field($fieldConfig) {
```
Comment (apathak18) [inline modules/acquia_cms_dam/acquia_cms_dam.module:54]: @rajeshreeputra Function parameters should be snake_case for example $field_config

File: modules/acquia_cms_dam/acquia_cms_dam.module [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * @file
+ * Contains hook implementations for ACMS DAM.
+ */
+
+/**
+ * Implements hook_modules_installed().
+ */
+function acquia_cms_dam_modules_installed($modules) {
```
Comment (apathak18) [inline modules/acquia_cms_dam/acquia_cms_dam.module:11]: Logic is repeated in this hook!!
@rajeshreeputra can we optimise logic created in this hook?
something like switch statements or create another method where you can put 
`// Update event image field to support Acquia DAM.
  if ($moduleHandler->moduleExists($module_name)) {
    $fieldConfig = \Drupal::configFactory()->getEditable($module_image_config);
    if ($fieldConfig) {
      _acquia_cms_dam_update_image_field($fieldConfig);
    }
  }`

Comment (rajeshreeputra) [reply modules/acquia_cms_dam/acquia_cms_dam.module:11]: Updated

File: modules/acquia_cms_dam/acquia_cms_dam.module (hunk 1 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_dam/acquia_cms_dam.module]: updated

**Review by apathak18 — APPROVED** [review]
Thanks 
Changes looks good to me.

**Review by chandan-singh7929 — APPROVED** [review]
Looks good to me, have added comment on the ticket please check.

File: modules/acquia_cms_article/config/pack_acquia_cms_article/cohesion_templates.cohesion_content_templates.node_article_full.yml [hunk 3]
```diff
@@ -3,11 +3,18 @@ langcode: en
...
+    - cohesion_custom_styles.cohesion_custom_style.65dc3072
+    - cohesion_custom_styles.cohesion_custom_style.7ef725fd
+    - cohesion_custom_styles.cohesion_custom_style.a13b8c94
+    - cohesion_custom_styles.cohesion_custom_style.bfd0fecd
+    - cohesion_custom_styles.cohesion_custom_style.ed4b7a3d
+    - cohesion_custom_styles.cohesion_custom_style.ee93d997
+    - cohesion_custom_styles.cohesion_custom_style.f3dc1bea
+    - cohesion_elements.cohesion_component.cpt_breadcrumbs
     - cohesion_templates.cohesion_master_templates.master_template
-    - image.style.coh_large_super_landscape
```
Comment (vishalkhode1) [inline modules/acquia_cms_article/config/pack_acquia_cms_article/cohesion_templates.cohesion_content_templates.node_article_full.yml:7]: @rajeshreeputra Why this image style dependencies are removed from here ? (Same for other configs).

File: modules/acquia_cms_event/config/optional/core.entity_view_display.node.event.horizontal_card.yml [hunk 4]
```diff
@@ -28,53 +28,61 @@ mode: horizontal_card
...
     type: text_summary_or_trimmed
-    weight: 2
-    region: content
     label: hidden
     settings:
       trim_length: 300
     third_party_settings: {  }
-  content_moderation_control:
-    weight: 0
+    weight: 2
```
Comment (vishalkhode1) [inline modules/acquia_cms_event/config/optional/core.entity_view_display.node.event.horizontal_card.yml:35]: @rajeshreeputra We are updating configurations, but we are missing update hooks. Can we please add those as well ? (Same for all configurations).

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: modules/acquia_cms_article/config/pack_acquia_cms_article/cohesion_templates.cohesion_content_templates.node_article_full.yml (hunk 3 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_article/config/pack_acquia_cms_article/cohesion_templates.cohesion_content_templates.node_article_full.yml:7]: image style dependency is automatically removed by site studio as now picture field is replaced by the drupal field.

File: modules/acquia_cms_event/config/optional/core.entity_view_display.node.event.horizontal_card.yml (hunk 4 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_event/config/optional/core.entity_view_display.node.event.horizontal_card.yml]: added update_hook

**Pull request closed without merging**